app/src/main/java/ch/epfl/sweng/erpa/operations/UserProfileCoordinator.java

```diff
@@ -0,0 +1,46 @@
+package ch.epfl.sweng.erpa.operations;
+
+import android.content.Intent;
+
+import com.annimon.stream.Optional;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import ch.epfl.sweng.erpa.model.UserProfile;
+
+import static ch.epfl.sweng.erpa.services.UserSignupService.createAccessToken;
+
+@Singleton
+public class UserProfileCoordinator implements InvocationHandler, DependencyCoordinator<UserProfile> {
```

Comment:
You can remove invocation handler, `DependencyCoordinator` extends `InvocationHandler`